core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -121,9 +128,178 @@ default ViewMetadata checkAndNormalize() {
+      this.changes.add(new MetadataUpdate.SetLocation(newLocation));
+      return this;
+    }
+
+    public Builder currentVersionId(int versionId) {
+      this.currentVersionId = versionId;
+      this.changes.add(new MetadataUpdate.SetCurrentViewVersion(versionId));
+      return this;
+    }
+
+    public Builder currentSchemaId(Integer schemaId) {
+      this.currentSchemaId = schemaId;
+      this.changes.add(new MetadataUpdate.SetCurrentSchema(schemaId));
+      return this;
+    }
+
+    public Builder addSchema(Schema schema) {
+      this.schemas.add(schema);
+      this.changes.add(new MetadataUpdate.AddSchema(schema, schema.highestFieldId()));
+      return this;
+    }
+
+    public Builder addAllSchemas(Iterable<Schema> schemaEntries) {
+      for (Schema schema : schemaEntries) {
+        this.schemas.add(schema);
+        this.changes.add(new MetadataUpdate.AddSchema(schema, schema.highestFieldId()));
+      }
+      return this;
+    }
+
+    public Builder schemas(Iterable<Schema> newSchemas) {
+      this.schemas = Lists.newArrayList(newSchemas);
+      for (Schema schema : newSchemas) {
+        this.changes.add(new MetadataUpdate.AddSchema(schema, schema.highestFieldId()));
+      }
+      return this;
+    }
+
+    public Builder addVersion(ViewVersion version) {
+      this.versions.add(version);
+      this.changes.add(new MetadataUpdate.AddViewVersion(version));
+      return this;
+    }
+
+    public Builder addAllVersions(Iterable<ViewVersion> versionEntries) {
+      for (ViewVersion version : versionEntries) {
+        this.versions.add(version);
+        this.changes.add(new MetadataUpdate.AddViewVersion(version));
+      }
+      return this;
+    }
+
+    public Builder versions(Iterable<ViewVersion> newVersions) {
+      this.versions = Lists.newArrayList(newVersions);
+      for (ViewVersion version : newVersions) {
+        this.changes.add(new MetadataUpdate.AddViewVersion(version));
+      }
+      return this;
+    }
+
+    public Builder addHistory(ViewHistoryEntry historyEntry) {
+      this.history.add(historyEntry);
+      return this;
+    }
+
+    public Builder addAllHistory(Iterable<ViewHistoryEntry> historyEntries) {
+      for (ViewHistoryEntry entry : historyEntries) {
+        this.history.add(entry);
+      }
+      return this;
+    }
+
+    public Builder history(Iterable<ViewHistoryEntry> newHistory) {
+      this.history = Lists.newArrayList(newHistory);
+      return this;
+    }
+
+    public Builder putProperties(String key, String value) {
+      this.properties.put(key, value);
+      this.changes.add(new MetadataUpdate.SetProperties(ImmutableMap.of(key, value)));
+      return this;
+    }
+
+    public Builder putAllProperties(Map<String, String> newProperties) {
+      this.properties.putAll(newProperties);
+      this.changes.add(new MetadataUpdate.SetProperties(newProperties));
+      return this;
+    }
+
+    public Builder properties(Map<String, String> newProperties) {
```

Comment:
I think this behavior is incorrect. Setting properties should not also remove properties.

Reply:
fixed